Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

```diff
@@ -1,46 +1,580 @@
+        "144", "AGGFRAG", "",
+        "145", "NSH", "",
+        "253", "", "",
+        "254", "", "",
+        "255", "Reserved", ""
+    ];
+    let seveirtylookupforpriority=datatable(priority: string, EventSeverity: string)[
+        "1", "High",
+        "2", "Medium",
+        "3", "Low",
+        "4", "Informational"
+    ];
+    let severitylookupforDvcAction=datatable(DvcAction: string, EventSeverity: string)[
+        "Allow", "Informational",
+        "Deny", "Low"
+    ];
+    let directionlookup=datatable(direction: string, NetworkDirection: string)[
+        "ingress", "Inbound",
+        "egress", "Outbound",
+        "Unknown", "Unknown"
+    ];
+    let parser=(disabled: bool=false) {
+        union isfuzzy=true
+            (
+            meraki_CL
+            | project-rename LogMessage =  Message
+            ),
+            (
+            Syslog
+            | project-rename LogMessage =  SyslogMessage
+            )
+        | where not(disabled) and (LogMessage has_any("flows", "firewall", "ids-alerts") or LogMessage has_all("security_event", "ids-alerted") or (LogMessage has "events" and (LogMessage has_any ("Blocked DHCP server response", "association") or (LogMessage has "VRRP packet" and not(LogMessage has_any ("VRRP passive", "VRRP active"))) or (LogMessage has "disassociation" and LogMessage !has "auth_neg_failed"))) or (LogMessage has "airmarshal_events" and LogMessage has_any("ssid_spoofing_detected", "rogue_ssid_detected")))
+        | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+        | extend
+            LogType = tostring(Parser[2]),
+            Substring = tostring(Parser[3])
+        | extend LogSubType = case( 
+                            LogType == "security_event",
+                            tostring(extract(@"^(\S+)\s", 1, Substring)),
+                            (LogType == "events" and Substring has "Blocked DHCP server response"),
+                            "Blocked DHCP server response",
+                            LogType == "events" and (Substring has "VRRP packet" and not(Substring in~ ("VRRP passive", "VRRP active"))),
+                            "Virtual router collision",
+                            LogType in~ ("events", "airmarshal_events"),
+                            extract(@"type[ ]{0,1}[=]{0,1}[ ]{0,1}(\S+)", 1, Substring),
+                            ""
+                        )
+        | where (LogType in("flows", "firewall", "ids-alerts") or (LogType == "security_event" and LogSubType == "ids-alerted") or (LogType == "events" and (LogSubType in ("Blocked DHCP server response", "association", "Virtual router collision") or (LogSubType has "disassociation" and LogSubType !has "auth_neg_failed"))) or (LogType == "airmarshal_events" and LogSubType in("ssid_spoofing_detected", "rogue_ssid_detected")))
+        | parse-kv Substring as(timestamp: string) with (pair_delimiter=" ", kv_delimiter="=")
```

Comment:
parse-kv should be used for more than one object. Single object should be solved using parse statement like this
| parse SubString with * "timestamp=" timestamp " " *

Reply:
Done